File: cpp/nth-prime/nth_prime.cpp
#include "nth_prime.h"
#include <cmath>
#include <stdexcept>

bool prime::isPrime(const int num) {
  for (int i = 2; i <= sqrt(num); i++) {
    if (num % i == 0) {
      return false;
    }
  }
  return true;
}

int prime::nth(const int n) {
  if (n == 0) {
    throw std::domain_error("invalid input");
  }
  int counter = 2;
  for (int i = 1; i <= n; i++) {
    while (!isPrime(counter++));
  }
  return counter - 1;
}

File: go/nth-prime/nth_prime.go
package prime

import "math"

// IsPrime checks if a number is prime.
func IsPrime(num int) bool {
	for i := 2; i <= int(math.Sqrt(float64(num))); i++ {
		if num%i == 0 {
			return false
		}
	}
	return true
}

// Nth returns the nth prime number
func Nth(n int) (int, bool) {
	if n == 0 {
		return 0, false
	}
	var primes = []int{2}
	for num := 3; len(primes) < n; num++ {
		if IsPrime(num) {
			primes = append(primes, num)
		}
	}
	return primes[len(primes)-1], true
}
